src/sql/src/plan.rs

```diff
@@ -1555,6 +1560,62 @@ impl ConnectionDetails {
     }
 }
 
+#[derive(Debug, Clone, Serialize, PartialEq, Eq, Ord, PartialOrd, Hash)]
+pub struct NetworkPolicyRule {
+    pub name: String,
+    pub action: NetworkPolicyRuleAction,
+    pub address: PolicyAddress,
+    pub direction: NetworkPolicyRuleDirection,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Ord, PartialOrd, Hash)]
+pub enum NetworkPolicyRuleAction {
+    Allow,
+}
+
+impl std::fmt::Display for NetworkPolicyRuleAction {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::Allow => write!(f, "allow"),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, PartialEq, Eq, Ord, PartialOrd, Hash)]
+pub enum NetworkPolicyRuleDirection {
+    Ingress,
+}
+
+impl std::fmt::Display for NetworkPolicyRuleDirection {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::Ingress => write!(f, "ingress"),
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Ord, PartialOrd, Hash)]
+pub struct PolicyAddress(pub IpNet);
+impl PolicyAddress {
+    pub fn to_string(&self) -> String {
+        self.0.to_string()
+    }
+}
+impl From<String> for PolicyAddress {
+    fn from(value: String) -> Self {
+        Self(IpNet::from_str(&value).expect("expected"))
+    }
+}
```

Comment:
It looks like right now this is only used in proto serialization, so the `.expect(...)` is fine, but for future proofing it feels like this should maybe be `impl TryFrom` or `impl std::str::FromStr`.

Reply:
👍  already done in the follow up pr
https://github.com/MaterializeInc/materialize/pull/30172/files#diff-1f1679d8d239da84a9ffbf2f13bd615849ae62d4f93fdf4e945b43bcd1e56606R1649